keycloakClient/keycloakClient.go

```diff
@@ -62,13 +62,38 @@ func New(config Config) (*Client, error) {
 	httpClient := http.Client{
 		Timeout: config.Timeout,
 	}
+	// Check if logging is enabled, if it is, pass in logger
+	if config.EnabledLogging {
+		loggingClient := LoggingClient{
+			standardClient:   httpClient,
+			StructuredLogger: config.Logger,
+		}
+		return &Client{
```

Comment:
don't return from the if, it makes the code harder to read. Create a pointer variable for Client and assign to it inside the if